plugin/kubernetes: AXFR crashes CoreDNS ("send on closed channel") when the zone has more than one NS address

### What happened

When the `transfer` plugin is enabled for a `kubernetes` zone, an AXFR/IXFR request crashes the **entire CoreDNS process** with `panic: send on closed channel` whenever CoreDNS has more than one nameserver address for the zone.

```
panic: send on closed channel

goroutine N [running]:
github.com/coredns/coredns/plugin/kubernetes.(*Kubernetes).Transfer.func1()
	.../plugin/kubernetes/xfr.go:50
```

The panic is raised in the background goroutine started in `Kubernetes.Transfer`. There is no `recover` on the transfer path (`plugin/transfer/transfer.go` just ranges the channel), so the panic terminates the whole process — a remotely-triggerable denial of service for any client permitted to request a zone transfer.

### Root cause

In `plugin/kubernetes/xfr.go`, the glue-address send (`ch <- nsAddrs`), the service transfer, the trailing SOA, and `close(ch)` are all nested **inside** the `for _, nsAddr := range nsAddrs` loop that should only emit one deduplicated `NS` record per nameserver hostname. On the first iteration the channel is closed; on the second iteration the goroutine sends on the closed channel and panics.

### How to reproduce / who is affected

`nsAddrs` returns one record per collected nameserver IP, and returns two or more in ordinary deployments:

- **Dual-stack CoreDNS Service** — two `ClusterIPs` (IPv4 + IPv6) → two glue records.
- **`!foundEndpoint` fallback** — when CoreDNS can't discover its own Service/Endpoint (running outside the cluster, or lacking RBAC to list its own endpoints), it falls back to one record per bound IP; a dual-stack pod (IPv4 + IPv6) or a host with two interfaces yields two records.

Any deployment that enables `transfer` for a kubernetes zone and matches one of the above crashes the moment a client issues `AXFR`/`IXFR`. It has been latent since #4696 because every test in `xfr_test.go` configures a single `localIP`, so the loop only ever runs once.

Secondary: if `nsAddrs` returns **zero** records the loop never runs, so `close(ch)` never runs and the consumer (`for range ch`) blocks forever (goroutine/connection leak). The same fix resolves it.

### Version

Reproduced on current `master` (`534f668a`); present since #4696 (CoreDNS 1.8.x).

### Fix

PR to follow — move the glue send, service transfer, trailing SOA, and `close(ch)` out of the per-nameserver loop so they run exactly once; only the deduplicated `NS`-record send stays inside.

--- a/plugin/kubernetes/xfr.go
+++ b/plugin/kubernetes/xfr.go
@@ -47,17 +47,17 @@ func (k *Kubernetes) Transfer(zone string, serial uint32) (<-chan []dns.RR, erro
 				nsHosts[nsHost] = struct{}{}
 				ch <- []dns.RR{&dns.NS{Hdr: dns.RR_Header{Name: zone, Rrtype: dns.TypeNS, Class: dns.ClassINET, Ttl: k.ttl}, Ns: nsHost}}
 			}
-			ch <- nsAddrs
-
-			if !k.isMultiClusterZone(zone) {
-				k.transferServices(ch, zonePath)
-			} else {
-				k.transferMultiClusterServices(ch, zonePath)
-			}
-
-			ch <- soa
-			close(ch)
 		}
+		ch <- nsAddrs
+
+		if !k.isMultiClusterZone(zone) {
+			k.transferServices(ch, zonePath)
+		} else {
+			k.transferMultiClusterServices(ch, zonePath)
+		}
+
+		ch <- soa
+		close(ch)
 	}()
 	return ch, nil
 }

--- a/plugin/kubernetes/xfr_test.go
+++ b/plugin/kubernetes/xfr_test.go
@@ -104,6 +104,65 @@ func TestKubernetesIXFRCurrent(t *testing.T) {
 	}
 }
 
+// TestKubernetesAXFRMultipleNSAddrs guards against a regression where the SOA
+// send, service transfer, and close(ch) were nested inside the per-NS-address
+// loop. When nsAddrs returns more than one record (e.g. a dual-stack CoreDNS
+// whose own Service/Endpoint is not discoverable, so nsAddrs falls back to one
+// record per bound IP), the second iteration sent on an already-closed channel
+// and crashed the process with "panic: send on closed channel".
+func TestKubernetesAXFRMultipleNSAddrs(t *testing.T) {
+	k := New([]string{"cluster.local."})
+	k.APIConn = &APIConnServeTest{}
+	k.Namespaces = map[string]struct{}{"testns": {}, "kube-system": {}}
+	// Dual-stack: two bound addresses -> nsAddrs returns two glue records that
+	// share the ns.dns.cluster.local. owner name.
+	k.localIPs = []net.IP{net.ParseIP("10.0.0.10"), net.ParseIP("fd00::10")}
+
+	ch, err := k.Transfer(k.Zones[0], 0)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	var xfr []dns.RR
+	for rrs := range ch {
+		xfr = append(xfr, rrs...)
+	}
+
+	if len(xfr) == 0 {
+		t.Fatal("no records returned from transfer")
+	}
+	if xfr[0].Header().Rrtype != dns.TypeSOA {
+		t.Errorf("transfer must start with SOA, got %s", dns.TypeToString[xfr[0].Header().Rrtype])
+	}
+	if xfr[len(xfr)-1].Header().Rrtype != dns.TypeSOA {
+		t.Errorf("transfer must end with SOA, got %s", dns.TypeToString[xfr[len(xfr)-1].Header().Rrtype])
+	}
+
+	var ns, a, aaaa int
+	for _, rr := range xfr {
+		switch rr.Header().Rrtype {
+		case dns.TypeNS:
+			ns++
+		case dns.TypeA:
+			if rr.Header().Name == "ns.dns.cluster.local." {
+				a++
+			}
+		case dns.TypeAAAA:
+			if rr.Header().Name == "ns.dns.cluster.local." {
+				aaaa++
+			}
+		}
+	}
+	// The NS owner name is deduplicated, so exactly one NS record is expected...
+	if ns != 1 {
+		t.Errorf("expected exactly 1 NS record, got %d", ns)
+	}
+	// ...while both nameserver glue addresses must be present.
+	if a != 1 || aaaa != 1 {
+		t.Errorf("expected both glue records (1 A, 1 AAAA) for ns.dns.cluster.local., got %d A and %d AAAA", a, aaaa)
+	}
+}
+
 func validateAXFR(t *testing.T, ch <-chan []dns.RR, multicluster bool) {
 	t.Helper()
 	xfr := []dns.RR{}

